rust_book_ex_8_3/src/vector_ex.rs

```diff
@@ -0,0 +1,28 @@
+use std::collections::HashMap;
+
+pub fn get_mean(v: Vec<u32>) -> f32{
+    let sum: u32 = v.iter().sum();
+    return sum as f32 / v.len() as f32;
+}
+
+pub fn get_median(v: Vec<u32>) -> u32 {
+    let size = v.len();
+    return v[size / 2];
+}
+
+pub fn get_mode(v: Vec<u32>) -> u32 {
+    let mut number_counter: HashMap<u32, u32> = HashMap::new();
+    let mut max_count_number = 0;
+    let mut max_count = 0;
+
+    for n in &v {
```

Comment:
This looks generally good (although typically variable names would be a bit shorter).

I think, though, if you vector is very long and there's many repeated values, it may be cheaper to first build the hashmap and then find its max element. E.g., if there's 1,000,000 elements and only 10 possible values, you'll have to do 10 comparisons instead of 1,000,000.

Also, for speed you could use a non-cryptographic hasher here, like fnv, via `HashMap::with_hasher()`.